charts/kube-prometheus-stack/templates/prometheus/serviceThanosSidecar.yaml

```
@@ -16,7 +16,9 @@ metadata:
 {{- end }}
 spec:
   type: {{ .Values.prometheus.thanosService.type }}
+  {{- if not .Values.prometheus.thanosService.omitClusterIP }}
   clusterIP: {{ .Values.prometheus.thanosService.clusterIP }}
```

As discussed in the linked tickets (see links in #863) regarding the nginx chart, the ideal solution here would be something like this, but if we want to maintain backward compatibility, we'll probably need something like `omitClusterIP`.

```suggestion
  {{- with .Values.prometheus.thanosService.clusterIP }}
  clusterIP: {{ . }}
  {{- end }}
```
```